src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -81,9 +85,41 @@ export class PasteImageProvider implements DocumentPasteEditProvider {
 			return;
 		}
 
-		widget.attachmentModel.addContext(imageContext);
+		// Make sure to attach only new contexts
+		const currentContextIds = widget.attachmentModel.getAttachmentIDs();
+		if (currentContextIds.has(imageContext.id)) {
+			return;
+		}
+
+		const customEdit = {
```

Comment:
Looks like some of this code is duplicated. Can you extract it a shared class or function?